src/rust/bitbox02-rust-c/src/async_usb.rs

```diff
@@ -0,0 +1,56 @@
+// Copyright 2020 Shift Cryptosecurity AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#[no_mangle]
+pub extern "C" fn rust_async_usb_spin() {
+    bitbox02_rust::async_usb::spin();
+}
+
+#[repr(C)]
+pub enum UsbResponse {
+    UsbResponseAck,
+    UsbResponseNotReady,
+    UsbResponseNack,
+}
+
+/// Polls for a result of an async usb task. If a result is available, it is copied to `out`.
+///
+/// Returns:
+/// `UsbResponseNack` if on ask is running.
+/// `UsbResponseAck` if the result was copied.
+/// `UsbResponseNotReady` if a task is running but not yet complete.
+#[no_mangle]
+pub unsafe extern "C" fn rust_async_usb_copy_response(out: *mut bitbox02::buffer_t) -> UsbResponse {
```

Comment:
Kind of weird to see `async` everywhere in the function names which don't return futures. Is there a `rust_*sync*_usb_copy_response`? If not, maybe simply name it `rust_usb_copy_response` here and similarily the others?

Reply:
the naming is just namespacing, as the `copy_response` is in the `rust` `async_usb` module.

Reply:
That's exactly my point. Why not cut the `async_` prefix? Then it would become `rust_usb_xxx`, right. In a future pull request, anyway.

Reply:
then also rename the file to `usb.rs` here and in the `bitbox02_rust` crate? I just put the async prefix there because it's a file dealing with task spawning, spinning, cancelling - all async things.